### PR TITLE
Fix issues #6356 on master

### PR DESCRIPTION
**Pull request by witem**
### Pull Request check-list

Closes: #6356
Backport: #7164 

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does your issue contain a link to existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Have you added an entry under `Future` in the changelog?

### Description of change

Fix ER_EMPTY_QUERY reject on update only virtual fields and add test. For master branch

### CODE REVIEW

**Comment by witem** [comment]
Strange error in CI build:
```
Error: ENOENT: no such file or directory, rename 'coverage/lcov.info' -> 'coverage/integration.info'
    at Error (native)
    at Object.fs.renameSync (fs.js:681:18)
    at [eval]:1:15
    at Object.exports.runInThisContext (vm.js:54:17)
    at Object.<anonymous> ([eval]-wrapper:6:22)
    at Module._compile (module.js:409:26)
    at node.js:595:27
    at nextTickCallbackWith0Args (node.js:436:9)
    at process._tickCallback (node.js:365:13)
```
I think it is not relevant to my changes

File: lib/model.js [hunk 1]
```diff
@@ -3445,7 +3445,8 @@ class Model {
         return instance.save(includeOptions).then(() => this[include.association.accessors.set](instance, {save: false, logging: options.logging}));
       });
     }).then(() => {
-      if (!options.fields.length) return this;
+      const withoutVirtualFields = options.fields.filter((field) => !this.constructor._isVirtualAttribute(field));
```
Comment (sushantdhiman) [inline lib/model.js]: `withoutVirtualFields` => `realFields`

File: lib/model.js [hunk 2]
```diff
@@ -3445,7 +3445,8 @@ class Model {
...
+      if (!options.fields.length || !withoutVirtualFields.length) return this;
```
Comment (sushantdhiman) [inline lib/model.js]: just `withoutVirtualFields.length` check should be sufficient

**Comment by witem** [comment]
changed

**Pull request merged**